Index: src/main/java/com/guerinet/stringparser/Test.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/guerinet/stringparser/Test.java	(revision )
+++ src/main/java/com/guerinet/stringparser/Test.java	(revision )
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2013-2015 Julien Guerinet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.guerinet.stringparser;
+
+/**
+ * @author Julien Guerinet
+ * @version 1.0
+ * @since 1.0
+ */
+public class Test {
+
+    public static void main(String[] args){
+        if(args.length == 0){
+            System.out.println("Args is empty");
+        }
+
+        System.out.println(args[0]);
+    }
+}
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build.gradle	(date 1431297639000)
+++ build.gradle	(revision )
@@ -16,7 +16,7 @@
  */
 
 apply plugin: 'java'
-apply plugin:'application'
+apply plugin: 'application'
 
 sourceCompatibility = 1.7
 version = '2.5'
@@ -40,4 +40,33 @@
 dependencies {
     compile 'com.squareup.okhttp:okhttp:2.3.0'
     compile 'net.sf.supercsv:super-csv:2.3.1'
-}
\ No newline at end of file
+}
+
+apply plugin: MobileStringParser
+
+configArgs{
+    url = "Blah"
+}
+
+class MobileStringParser implements Plugin<Project> {
+    void apply(Project project) {
+        project.extensions.create('configArgs', MobileStringParserPluginExtension)
+        project.configArgs.extensions.create('languageArgs',LanguagePluginExtension)
+        project.task( 'parseStrings', type: JavaExec) {
+            main = 'com.guerinet.stringparser.Test'
+            classpath = project.sourceSets.main.runtimeClasspath
+            args 'Blah'
+        }
+    }
+}
+
+class MobileStringParserPluginExtension{
+    String url = null
+    String platform = null
+    List<LanguagePluginExtension> languagePluginExtensions
+}
+
+class LanguagePluginExtension{
+    String languageId = null
+    String languagePath
+}
\ No newline at end of file
